lib/routes/shared/dev/dev.dart

```diff
@@ -282,4 +291,83 @@ class DevViewState extends State<DevView> {
     bloc.accountSettingsSink.add(settings.copyWith(showConnectProgress: !settings.showConnectProgress));
   }
+
+  void _gotoMarketplaceScreen() async {
+    _promptPassword();
+  }
+
+  Future _promptPassword() {
+    return showDialog<bool>(
+        context: context,
+        barrierDismissible: false,
+        builder: (BuildContext context) {
+          return new AlertDialog(
+            contentPadding: EdgeInsets.only(left: 24.0, right: 24.0),
+            title: new Text(
+                "Enter credentials", style: theme.alertTitleStyle),
+            content: new SingleChildScrollView(
+                child: Form(
+                  key: _formKey,
+                  child: new Container(
+                    child: new TextFormField(
+                      obscureText: true,
+                      decoration:
+                      new InputDecoration(
+                        hintText: "Password",
+                        hintStyle: theme.alertStyle,
+                        focusedBorder: UnderlineInputBorder(
+                            borderSide: BorderSide(
+                                color: Colors.black45,
+                                style: BorderStyle.solid,
+                                width: 2.0)),),
+                      style: theme.alertStyle,
+                      textCapitalization: TextCapitalization.none,
+                      validator: (value) {
+                        if (value.isEmpty) {
+                          return "Please enter password.";
+                        } else if (!_validatePassword(
+                            value)) {
+                          return "Invalid password.";
+                        }
+                      },
+                    ),
+                  ),)
+            ),
+            actions: <Widget>[
+              new FlatButton(
+                child: new Text("Cancel", style: theme.cancelButtonStyle),
+                onPressed: () {
+                  Navigator.of(context).pop(false);
+                },
+              ),
+              new FlatButton(
+                child: new Text("OK", style: theme.buttonStyle),
+                onPressed: () {
+                  if (_formKey.currentState.validate()) {
+                    Navigator
+                        .of(_scaffoldKey.currentState.context)
+                        .pushReplacementNamed("/marketplace");
+                  }
+                },
+              ),
+            ],
+          );
+        });
+  }
+
+  bool _validatePassword(String value) {
+    if (value == _marketplacePassword) {
+      return true;
+    }
+    return false;
+  }
+
+  Future _getMarketplacePassword() async {
+    String configString = await rootBundle.loadString('conf/breez.conf');
+    Config config = Config.fromString(configString);
+    setState(() {
+      _marketplacePassword =
+          config.get("Application Options", "marketplacepassword");
+    });
+  }
```

Comment:
@roeierez Where should we move this? AccountBloc?

Reply:
> Where should we move this? AccountBloc?

No, to me it feels the right place.